includes/EnvoimoinscherModel.php

```diff
@@ -278,26 +278,135 @@ public function getEligibleOrdersCount()
 				 ON eo.'._DB_PREFIX_.'orders_id_order = o.id_order
 			 JOIN '._DB_PREFIX_.'carrier c
 				 ON c.id_carrier = oc.id_carrier
+			 JOIN '._DB_PREFIX_.'address a
+				 ON a.id_address = o.id_address_delivery
+			 JOIN '._DB_PREFIX_.'customer cr
+				 ON cr.id_customer = a.id_customer
 			 LEFT JOIN '._DB_PREFIX_.'emc_orders_errors er
 				 ON er.'._DB_PREFIX_.'orders_id_order = o.id_order
-			 WHERE (c.external_module_name = "envoimoinscher" OR o.delivery_number = 0)
-				 AND eo.ref_emc_eor IS NULL
-			 GROUP BY o.id_order
-			 ORDER BY o.id_order DESC';
+			 WHERE eo.ref_emc_eor IS NULL';
 
-		$orders = $this->db->ExecuteS($sql);
-		//we sort orders by EMC/other carriers
-		$final = array('emc' => 0, 'others' => 0, 'errors' => 0);
-		foreach ($orders as $order)
+		//apply filters
+		if (!empty($params['filterBy']))
 		{
-			$key = 'others';
-			if ($order['errors_eoe'] != '')
-				$key = 'errors';
-			elseif ($order['external_module_name'] == 'envoimoinscher')
-				$key = 'emc';
-			$final[$key]++;
+
+			//by order type
+			if (isset($params['filterBy']['type_order']))
+			{
+				switch ($params['filterBy']['type_order'])
+				{
+					case 0:
+						$sql .= ' AND c.external_module_name = "envoimoinscher" AND (er.errors_eoe = "" OR er.errors_eoe is NULL)';
+						break;
+					case 1:
+						$sql .= ' AND c.external_module_name != "envoimoinscher" AND (er.errors_eoe = "" OR er.errors_eoe is NULL)';
+						break;
+					case 2:
+						$sql .= ' AND er.errors_eoe != ""';
+						break;
+					default:
+						break;
+				}
+			}
+
+			//by order id
+			if (isset($params['filterBy']['filter_id_order']))
+				$sql .= ' AND o.id_order = '.(int)$params['filterBy']['filter_id_order'];
+
+			//by order status
+			if (isset($params['filterBy']['status']) && is_array($params['filterBy']['status']))
+			{
+				$filter = ' AND o.current_state IN (';
+				$statuses = array();
+				foreach ($params['filterBy']['status'] as $value)
+					array_push($statuses, $value);
+				$filter .= implode(',', $statuses).')';
+				if (count($statuses) > 0)
+					$sql .= $filter;
```

Comment:
```php
	if (count($statuses) > 0)
	{
		$filter .= implode(',', $statuses).')';
		$sql .= $filter;
	}
```

I add this line in the test, because its execution is useless if ```$statuses``` is empty.